unit_tests/library/route.cpp

```diff
@@ -22,15 +22,124 @@ BOOST_AUTO_TEST_CASE(test_route)
     using namespace osrm;
 
     RouteParameters params;
-
+    params.coordinates.emplace_back(util::FloatLongitude{}, util::FloatLatitude{});
     params.coordinates.emplace_back(util::FloatLongitude{}, util::FloatLatitude{});
     params.coordinates.emplace_back(util::FloatLongitude{}, util::FloatLatitude{});
 
     json::Object result;
-
     const auto rc = osrm.Route(params, result);
+    BOOST_CHECK(rc == Status::Ok);
```

Comment:
Try to do these checks fixture based. That means you need:

1. Code that just creates the full expected response as util::json::Object (hard coded)
2. Code that validates if the response and the given object are the same

Hard coding in tests is exactly what you want. There should be as little actual code as possible to minimize surprises. Tests need to be as stupid as possible.
I feel a lot of the tests that I wrote in the past that included extensive clever setup code are pretty bad and break often.